plugin-elasticsearch/src/api/client.rs

```diff
@@ -0,0 +1,173 @@
+    pub fn new(conn: ConnectionSettings, data: DataSettings) -> anyhow::Result<Self> {
+        let mut builder = reqwest::blocking::ClientBuilder::new();
+        if conn.allow_insecure {
+            builder = builder
+                .danger_accept_invalid_certs(true)
+                .danger_accept_invalid_hostnames(true);
+        }
+        builder = builder
+            .user_agent(user_agent_string())
+            .default_headers(HeaderMap::from_iter(vec![(
+                reqwest::header::AUTHORIZATION,
+                conn.auth
+                    .into_header_value()
+                    .context("auth should make a valid http header")?,
+            )]));
+        let client = builder.build().context("failed to initialize HTTP client")?;
+        Ok(Self {
+            client,
+            server_url: conn.server_url,
+            serializer: Serializer {
+                index_prefix: data.index_prefix,
+                metric_unit_as_index_suffix: data.metric_unit_as_index_suffix,
+            },
+        })
+    }
+
+    pub fn create_index_template(&self) -> anyhow::Result<()> {
+        const TEMPLATE_NAME: &str = "alumet_index_template";
+
+        let template = IndexTemplate {
+            mappings: self.serializer.common_index_mappings(),
+        };
+        let index_pattern = format!("{}-*", self.serializer.index_prefix);
+        let create = CreateIndexTemplate {
+            index_patterns: vec![index_pattern],
+            template,
+            priority: 80,
+            version: 3,
+            meta: HashMap::from_iter([("origin".to_string(), "Alumet measurements".to_string())]),
+        };
+
+        // Create the template (or update it if it exists).
+        // Note: server_url is turned into a string with a trailing '/'.
+        let url = format!("{}_index_template/{TEMPLATE_NAME}", self.server_url);
+        let request = self.client.put(url).json(&create);
+        log::trace!("sending {request:?}");
+
+        let response = request.send().context("could not send request")?;
+        log::trace!("got response {response:?}");
+
+        Self::handle_response(response)?;
+        Ok(())
+    }
+
+    pub fn bulk_insert_measurements(&self, m: &MeasurementBuffer, ctx: &OutputContext) -> anyhow::Result<()> {
+        let url = self.server_url.join("_bulk").unwrap();
+
+        let body = self
+            .serializer
+            .body_bulk_create_docs(m, ctx)
+            .context("measurements serialization failed")?;
+
+        log::trace!("serialized measurements:\n{body}");
+        let request = self.client.put(url).body(body).header(
+            reqwest::header::CONTENT_TYPE,
+            HeaderValue::from_static("application/x-ndjson"),
+        );
+        log::trace!("sending {request:?}");
+        let response = request.send().context("could not send request")?;
+
+        log::trace!("got response {response:?}");
+        Self::handle_response(response)?;
+        // let failed_items = response
```

Comment:
let's remove comment ?

Reply:
oops! Yes it should definitely be removed